### DISCUSSION TITLE
ESP32 Data streaming to Desktop Interface via Husarnet

### DISCUSSION THREAD

**Discussion by McStick0** [jameszah/ESP32-CAM-Video-Telegram#2] [Q&A]
Hi!
Hi, I'm working on a personal Hobby project 😄 
Necessary things:
- ESP32 Wrover-B
- ESP32-CAM
- SIM7600E(in future)
- Husarnet
- Interface

Streaming data (eg video, commands to GPIO) via internet, together with an Interface that will decode
data and display as well as send data to Esp32.
I thought to save esp32 resources and reduce latency, render video will be done in desktop application.

![Progress map](https://user-images.githubusercontent.com/86937384/124408281-2994d100-dd46-11eb-9b11-d0d89dc98e85.jpg)
- [ ] 1. Send  data to second ESP32
- [ ] 2. Reciving and send data to Husarnet
- [ ] 3. Reciving, decode, show data on Interface. 

The question is, has anyone ever worked on a similar project?
Look for tips, tutorials, pages on a few dilemmas such as:

1. what to use for communication between esp-esp and esp-interface?
2. how to "catch" this data from ESP and process it in an interface - is it possible?

**Comment by jameszah** [comment]
Have you seen the That Project videos https://www.youtube.com/c/ThatProject/featured

He has a bunch of videos of streaming esp32-cam video to different devices, and through flutter and mqtt to multiple devices.

One problem I had with the basic ESP32-CAM recording to sd and displaying the video on a small color screen at the same time ... is that you run out of spi interfaces.  You can use an i2s screen but not with the big throughout.

Don't know if he has added security through a Husarnet or equivalent.
https://www.youtube.com/watch?v=aY7i0xnQW54&t=746s
